content/blog/2019/11/2019-11-21-welcome-to-the-matrix.adoc

```diff
@@ -0,0 +1,389 @@
+                    exclude {
+                        axis {
+                            name 'PLATFORM'
+                            values 'linux'
+                        }
+                        axis {
+                            name 'BROWSER'
+                            values 'safari'
+                        }
+                    }
+                    exclude {
+                        axis {
+                            name 'PLATFORM'
+                            notValues 'windows'
+                        }
+                        axis {
+                            name 'BROWSER'
+                            values 'edge'
+                        }
+                    }
+                }
+                stages {
+                    stage('Build') {
+                        steps {
+                            echo "Do Build for ${PLATFORM} - ${BROWSER}"
+                        }
+                    }
+                    stage('Test') {
+                        steps {
+                            echo "Do Test for ${PLATFORM} - ${BROWSER}"
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+----
+
+
+
+.Log output (truncated - PLATFORM_FILTER = 'mac' )
+[source]
+----
+...
+[Pipeline] stage
+[Pipeline] { (BuildAndTest)
+[Pipeline] parallel
+[Pipeline] { (Branch: Matrix - OS = 'linux', BROWSER = 'firefox')
+[Pipeline] { (Branch: Matrix - OS = 'windows', BROWSER = 'firefox')
+[Pipeline] { (Branch: Matrix - OS = 'mac', BROWSER = 'firefox')
+[Pipeline] { (Branch: Matrix - OS = 'linux', BROWSER = 'chrome')
+[Pipeline] { (Branch: Matrix - OS = 'windows', BROWSER = 'chrome')
+[Pipeline] { (Branch: Matrix - OS = 'mac', BROWSER = 'chrome')
+[Pipeline] { (Branch: Matrix - OS = 'windows', BROWSER = 'safari')
+[Pipeline] { (Branch: Matrix - OS = 'mac', BROWSER = 'safari')
+[Pipeline] { (Branch: Matrix - OS = 'windows', BROWSER = 'edge')
+...
+Stage "Matrix - OS = 'linux', BROWSER = 'chrome'" skipped due to when conditional
+Stage "Matrix - OS = 'linux', BROWSER = 'firefox'" skipped due to when conditional
+...
+Do Build for mac - firefox
+Do Build for mac - chrome
+Do Build for mac - safar
+...
+Stage "Matrix - OS = 'windows', BROWSER = 'chrome'" skipped due to when conditional
+Stage "Matrix - OS = 'windows', BROWSER = 'edge'" skipped due to when conditional
+...
+Do Test for mac - safari
+Do Test for mac - firefox
+Do Test for mac - chrome
+----
+
+
+== Conclusion 
+
+In this blog post, we've looked at how to use the `matrix` directive to make concise but powerful declarative pipelines.  An equivalent pipeline created without `matrix` would easily be several times larger, and much harder to understand and maintain. 
+
+Matrix is now available from the experimental update center. 
+It will be released to the main update center as soon as we're done putting the finishing touches on the documentation and online help. 
+
+TODO: Join me at DWJW in Lisbon.
```

Comment:
Are you going to give a talk about Matrix at DW|JW? If so, I'd put in something to the effect of

=== Join me at DevOps World | Jenkins World in Lisbon!
I'll be giving a talk in the Shangri-La conference room, at 3:45AM on Tuesday night. Hope to see you there!

And maybe move it up above the Conclusion, since the Conclusion should probably be the last thing.